blaze/compute/core.py

```diff
@@ -446,10 +446,9 @@ def compute(expr, d, **kwargs):
     optimize_ = kwargs.get('optimize', optimize)
     pre_compute_ = kwargs.get('pre_compute', pre_compute)
     post_compute_ = kwargs.get('post_compute', post_compute)
-
     expr2, d2 = swap_resources_into_scope(expr, d)
     if pre_compute_:
-        d3 = dict([(e, pre_compute_(expr2, dat, **kwargs))
+        d3 = dict([(e, pre_compute_(e, dat, **kwargs))
```

Comment:
This is what was causing the wrong shape to be applied to the rest of the data types, can someone make sure that this is the correct fix